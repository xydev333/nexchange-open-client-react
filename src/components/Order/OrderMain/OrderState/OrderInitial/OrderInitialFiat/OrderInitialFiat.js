import React from 'react';
import { I18n } from 'react-i18next';
import styles from '../OrderInitial.scss';

const OrderInitial = props => {
  return (
   <I18n ns="translations">
	{(t) => (
    <div id="order-payment" className={`row ${styles.container}`}>
      <div id="order-payment-details" className="col-xs-12 col-ms-6 col-sm-6 col-md-4">
        <h3>
          {t('order.initial1')}:{' '}
          <span className={styles.time}>
            <b>{props.time}</b>
          </span>
        </h3>
        <h4>
          {t('order.pay')}{' '}
          <b>
            {parseFloat(props.order.amount_quote)} {props.order.pair.quote.code}
          </b>
        </h4>
        <a href={props.order.payment_url} target="_blank" className="btn btn-default btn-lg">
          <i className="fas fa-credit-card" aria-hidden="true" style={{ position: 'relative', left: -13 }} />
          {t('order.fiat.status.pay')}
        </a>
      </div>

      <div className={`col-xs-12 col-ms-6 col-sm-6 col-md-8 ${styles.cards}`}>
        <h3>{t('order.fiat.cards')}:</h3>

        <div className="visible-xs-block visible-sm-block">
          <img src="/img/order/cards-mobile.png" alt={t('order.fiat.cardsaccepted')} />
        </div>

        <div className="visible-md-block visible-lg-block">
          <img src="/img/order/cards-desktop.png" alt={t('order.fiat.cardsaccepted')} />
        </div>
      </div>
    </div>
	)}
   </I18n>
  );
};

export default OrderInitial;
