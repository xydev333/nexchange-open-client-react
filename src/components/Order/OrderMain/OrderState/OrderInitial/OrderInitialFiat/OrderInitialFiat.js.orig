import React from 'react';
<<<<<<< HEAD:src/components/Order/OrderMain/OrderState/OrderInitial/OrderInitialFiat/OrderInitialFiat.js
import styles from '../OrderInitial.scss';

const OrderInitial = props => {
  return (
    <div id="order-payment" className={`row ${styles.container}`}>
      <div id="order-payment-details" className="col-xs-12 col-ms-6 col-sm-6 col-md-4">
        <h3>
          Time remaining:{' '}
          <span className={styles.time}>
=======
import { I18n } from 'react-i18next';

const OrderInitial = props => {
  return (
   <I18n ns="translations">
	{(t) => (
    <div id="order-payment" className="row">
      <div id="order-payment-details" className="col-xs-12 col-ms-6 col-sm-6 col-md-4">
        <h3>
          {t('order.initial1')}:{' '}
          <span id="time-remaining">
>>>>>>> Translated further components:src/containers/order-fiat/OrderInitial.js
            <b>{props.time}</b>
          </span>
        </h3>
        <h4>
          {t('order.pay')}{' '}
          <b>
            {parseFloat(props.order.amount_quote)} {props.order.pair.quote.code}
          </b>
        </h4>
        <a href={props.order.payment_url} target="_blank" className="btn btn-default btn-lg">
<<<<<<< HEAD:src/components/Order/OrderMain/OrderState/OrderInitial/OrderInitialFiat/OrderInitialFiat.js
          <i className="fas fa-credit-card" aria-hidden="true" style={{ position: 'relative', left: -13 }} />
          Pay now
        </a>
      </div>

      <div className={`col-xs-12 col-ms-6 col-sm-6 col-md-8 ${styles.cards}`}>
        <h3>Accepted cards:</h3>
=======
          <i className="fa fa-credit-card" aria-hidden="true" style={{ position: 'relative', left: -13 }} />
          {t('order.fiat.status.pay')}
        </a>
      </div>

      <div id="cards" className="col-xs-12 col-ms-6 col-sm-6 col-md-8">
        <h3>{t('order.fiat.cards')}</h3>
>>>>>>> Translated further components:src/containers/order-fiat/OrderInitial.js

        <div className="visible-xs-block visible-sm-block">
          <img src="/img/order/cards-mobile.png" alt={t('order.fiat.cardsaccepted')} />
        </div>

        <div className="visible-md-block visible-lg-block">
          <img src="/img/order/cards-desktop.png" alt={t('order.fiat.cardsaccepted')} />
        </div>
      </div>
    </div>
	)}
   </I18n>
  );
};

export default OrderInitial;
