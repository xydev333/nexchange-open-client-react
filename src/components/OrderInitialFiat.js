import React, { Component } from 'react';
import {Icon} from 'react-fa';
import CopyToClipboard from 'react-copy-to-clipboard';

import CreditCardModal from './CreditCardModal';


class OrderInitial extends Component {
	constructor(props) {
		super(props);

		this.state = {
			showCreditCardModal: false
		}
	}

	triggerCopyTooltip() {
		$('#copy-to-clipboard').tooltip({
			trigger: 'click',
			placement: 'top'
		});

		$('#copy-to-clipboard').tooltip('hide')
			.attr('data-original-title', 'Wallet address copied!')
			.tooltip('show');

		setTimeout(() => {
			$('#copy-to-clipboard').tooltip('destroy');
		}, 1000);
	}

	render() {
	    return (
	    	<div id="order-payment">
    			<div className="col-xs-12 col-sm-4 col-md-3">
    				<img src="" />
    			</div>

    			<div id="order-payment-details" className="col-xs-12 col-sm-8 col-md-9">
    				<h3>Time remaining: <span id="time-remaining"><b>{this.props.timeRemaining}</b></span></h3>

    				<h4>Pay <b>{this.props.depositAmount} {this.props.depositCoin}</b></h4>

					<button type="button" className="btn btn-default btn-themed" onClick={() => this.setState({showCreditCardModal: true})}>
						<i className="fa fa-credit-card" aria-hidden="true" style={{position: "relative", left: -13}}></i>
						Pay now
					</button>
    			</div>

			    <CreditCardModal show={this.state.showCreditCardModal} onClose={() => this.setState({showCreditCardModal: false})} />
		    </div>
	    );
	}
}

export default OrderInitial;
