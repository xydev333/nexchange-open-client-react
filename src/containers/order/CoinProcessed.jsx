import React, { Component } from 'react';
import Helpers from '../../helpers';

class CoinProcessed extends Component {
  constructor(props) {
    super(props);
    this.state = { order: this.props.order };
    this.prepareState = this.prepareState.bind(this);
  }

  componentDidMount() {
    this.prepareState(this.props);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ order: nextProps.order }, () => {
      this.prepareState(nextProps);
    });
  }

  prepareState(props) {
    if (props.type === 'Deposit') {
      this.setState({
        coin: props.order.pair.quote.code,
        oppositeCoin: props.order.pair.base.code,
        amount: parseFloat(props.order.amount_quote),
        address: props.order.deposit_address
          ? props.order.deposit_address.address
          : '',
        order: props.order,
      });
    } else if (props.type === 'Receive') {
      this.setState({
        coin: props.order.pair.base.code,
        oppositeCoin: props.order.pair.quote.code,
        amount: parseFloat(props.order.amount_base),
        address: props.order.withdraw_address
          ? props.order.withdraw_address.address
          : '',
        order: props.order,
      });
    }
  }

  render() {
    let rates = ``;

    if (this.state.order && this.state.order.price) {
      rates += `Rates at order creation: \n`;
      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${(1 / this.state.order.price.rate).toFixed(8)} ${
          this.state.oppositeCoin
        }\n`;
      else if (this.props.type === 'Receive')
        rates += `${this.state.order.price.rate.toFixed(8)} ${
          this.state.oppositeCoin
        }\n`;

      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${(
          1 /
          this.state.order.price.rate *
          this.state.order.price.rate_usd
        ).toFixed(8)} USD\n`;
      else if (this.props.type === 'Receive')
        rates += `${this.state.order.price.rate_usd.toFixed(8)} USD\n`;

      rates += `1 ${this.state.coin} = `;

      if (this.props.type === 'Deposit')
        rates += `${(
          1 /
          this.state.order.price.rate *
          this.state.order.price.rate_btc
        ).toFixed(8)} BTC`;
      else if (this.props.type === 'Receive')
        rates += `${this.state.order.price.rate_btc.toFixed(8)} BTC`;

      if (
        this.state.order.user_provided_amount === 1 &&
        this.props.type === 'Receive'
      ) {
        rates += `\n\nWithdrawal fee: \n`;
        rates += `${this.state.order.withdrawal_fee} ${
          this.state.order.pair.base.code
        }`;
      } else if (
        this.state.order.user_provided_amount === 0 &&
        this.props.type === 'Deposit'
      ) {
        rates += `\n\nWithdrawal fee: \n`;
        rates += `${this.state.order.withdrawal_fee_quote} ${
          this.state.order.pair.quote.code
        }`;
      }
    }

    return (
      <div className="col-xs-12 col-sm-6">
        <div
          className={`coin-box box media ${
            this.props.type === 'Deposit' &&
            Helpers.isFiatOrder(this.props.order)
              ? 'fiat'
              : ''
          }`}
        >
          <div className="media-left">
            <i
              className={`coin-icon cc-${this.state.coin} ${this.state.coin}`}
            />
          </div>

          <div className="media-body">
            <h5>
              <b>
                {this.props.type} {this.state.amount} {this.state.coin}
              </b>

              <i
                className="fa fa-question-circle"
                data-toggle="tooltip"
                data-placement="top"
                style={{ marginLeft: 8 }}
                data-original-title={rates}
              />
            </h5>
            <h6>{this.state.address}</h6>
          </div>
        </div>
      </div>
    );
  }
}

export default CoinProcessed;
