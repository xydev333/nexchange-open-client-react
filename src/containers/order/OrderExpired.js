import React, { Component } from 'react';

class OrderExpired extends Component {
	componentDidMount() {
		$('[data-toggle="tooltip"], [rel="tooltip"]').tooltip();
	}

	render() {
	    return (
			<div className="text-center">
				<h2>Order Expired</h2>
				<h5>Your order is expired. You can make a new exchange order.</h5>
				
				<div>
					<h4
						className="text-warning"
						data-toggle="tooltip"
						data-placement="top"
						data-original-title="If you have already sent your coins, donâ€™t worry. Once we get them, the status of the order will be updated retroactively.">
						Already sent?
					</h4>
				</div>
			</div>
	    );
	}
}

export default OrderExpired;
