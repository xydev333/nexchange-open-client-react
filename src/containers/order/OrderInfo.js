import React from 'react';

import '../../css/order-crypto.scss';
import '../../css/order-fiat.scss';

import OrderInitial from './OrderInitial';
import OrderPreReleased from './OrderPreReleased';
import OrderReleased from './OrderReleased';
import OrderSuccess from './OrderSuccess';
import OrderFailure from './OrderFailure';
import OrderRefunded from './OrderRefunded';

import OrderPaymentFiat from '../order-fiat/OrderPayment';
import OrderPaidFiat from '../order-fiat/OrderPaid';
import OrderStatusFiat from '../order-fiat/OrderStatus';

import OrderPaymentCrypto from '../order-crypto/OrderPayment';
import OrderPaidCrypto from '../order-crypto/OrderPaid';
import OrderStatusCrypto from '../order-crypto/OrderStatus';

import STATUS_CODES from '../../statusCodes';
import Helpers from '../../helpers';

const OrderInfo = props => {
  const isFiatOrder = Helpers.isFiatOrder(props.order);
  let order;

  switch (STATUS_CODES[props.order.status_name[0][0]]) {
    case 'INITIAL':
      order = <OrderInitial {...props} />;
      break;
    case 'PAID_UNCONFIRMED':
      order = isFiatOrder ? (
        <OrderPaymentFiat {...props} />
      ) : (
        <OrderPaymentCrypto {...props} />
      );
      break;
    case 'PAID':
      order = isFiatOrder ? (
        <OrderPaidFiat {...props} />
      ) : (
        <OrderPaidCrypto {...props} />
      );
      break;
    case 'PRE_RELEASE':
      order = <OrderPreReleased {...props} />;
      break;
    case 'RELEASE':
      order = <OrderReleased {...props} />;
      break;
    case 'COMPLETED':
      order = <OrderSuccess {...props} />;
      break;
    case 'CANCELLED':
      order = <OrderFailure {...props} />;
      break;
    case 'REFUNDED':
      order = <OrderRefunded {...props} />;
      break;
    default:
      order = (
        <div className="text-center">
          <h2>Unknown order state, something went wrong</h2>
        </div>
      );
  }

  return (
    <div className="col-xs-12">
      <div className="box">
        {order}

        {isFiatOrder ? (
          <OrderStatusFiat status={props.order.status_name[0][0]} />
        ) : (
          <OrderStatusCrypto status={props.order.status_name[0][0]} />
        )}
      </div>
    </div>
  );
};

export default OrderInfo;
