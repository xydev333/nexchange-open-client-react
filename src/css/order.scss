@import "variables";
@import "mixins";
@import "bootstrap/variables";
@import "bootstrap/mixins";

#order { background: #f8f7f8; padding: 15px 0 0 0;
	a { color: $green; 
		&:visited, &:focus { text-decoration: none; }
		&:hover { text-decoration: underline; }
		&.btn { color: #fff;
			&:hover { text-decoration: none; }
		}
	}
	#order-ref { margin-bottom: 15px; }
	.box {
		h4 { font-size: 15px; }
		h5, h6 { margin: 0; text-transform: none; }
		.coin-icon { position: relative; top: 8px; font-size: 28px;
			@media (min-width: $screen-sm-min) { top: 9px; }
			@media (min-width: $screen-md-min) { top: 11px; }
		}
		.text-warning { color: $green; }		

		@media (min-width: $screen-sm-min) { margin-bottom: 30px; }
	}

	&.order-fiat {
		.box.fiat {
			h5 { margin-top: 11px; padding-bottom: 7px;
				@media (min-width: $screen-sm-min) { margin-top: 11px; padding-bottom: 13px; }
			}
		}
	}

	.coin-box { padding-left: 10px;
		.tooltip-inner { text-align: left; }
	}
	#time-remaining { color: #2cb4a0; }
	.order-status-section { 
		h2 { margin-top: 0; }
		.explanation { font-size: 16px; margin-top: 15px; }
	}


	@media (min-width: $screen-sm-min) { padding: 25px 0;
		h5 { font-size: 15px; }
		h6 { font-size: 12px; }
		.box h4 { font-size: 18px; }
	}

	@media (min-width: $screen-md-min) {
		h5 { font-size: 17px; }
		h6 { font-size: 16px; }
	}
}

#order-header {
	#bookmark-button { font-size: 20px; padding: 0; margin-top: 0; color: $green; }

	@media (min-width: $screen-ms-min) {
		h3 { float: left; }
		#bookmark-button { float: right; margin-top: 22px; }
	}
}

#order-payment { text-align: center;
	h3 { margin-top: 0; }
	@media (min-width: $screen-ms-min) { text-align: left; }
}

#bookmark {
	.bookmark-text{ font-size: 19px; }
	.input-group-addon { cursor: pointer; padding-top: 0;
		i { font-size: 18px; }
	}
	.form-group { margin-top: 10px; }
	input { font-size: 18px; }
}

.order-status-section {
	h2 { font-size: 24px; }

	@media (min-width: $screen-sm-min) {
		h2 { font-size: 30px; }
	}
}

#share-referral, #notifications { text-align: center;
	.box { padding: 20px; }
	.share { margin-top: 15px; }
	.fa { font-size: 45px; margin: 0 15px 0 0; color: #d6d6d6; }
	h2 { margin-top: 0; font-size: 22px;
		@media (min-width: $screen-sm-min) { font-size: 26px; }
		@media (min-width: $screen-md-min) { font-size: 34px; }
	}
}

#notifications {
	.form-group { margin-top: 10px; }
	button { width: 100%; margin-bottom: 0; }
	form { padding-top: 15px;
		h4 { position: absolute; top: -15px; left: 50%; transform: translate(-50%, 0); }
	}
	@media (min-width: $screen-sm-min) { button { width: auto; } }
}

@include keyframes(color-change) {
	from { color: #adadad; }
	to { color: #fbc02d; } 
}

@import "order-status";
@import "referral";
