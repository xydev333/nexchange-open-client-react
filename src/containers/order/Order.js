import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchOrder } from '../../actions';

import isFiatOrder from '../../helpers/isFiatOrder';
import config from '../../config';

import '../../css/order.scss';

import OrderInfo from './OrderInfo';
import OrderTop from './OrderTop';
import CoinProcessed from './CoinProcessed';

import NotFound from '../../components/NotFound';
import Loading from '../../components/Loading';

import Notifications from '../../containers/Notifications';
import RefundAddress from '../../containers/RefundAddress';
import ReferralBox from '../../containers/ReferralBox';

class Order extends Component {
  constructor(props) {
    super(props);

    if (this.props.order && this.props.match.params.orderRef === this.props.order.unique_reference) {
      this.state = { order: this.props.order };
    } else {
      this.state = {};
    }
  }

  componentDidMount() {
    this.props.fetchOrder(this.props.match.params.orderRef);
  }

  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      clearTimeout(this.timeout);
      clearInterval(this.interval);
      this.props.fetchOrder(this.props.match.params.orderRef);
    }
  }

  componentWillUnmount() {
    clearInterval(this.interval);
    clearTimeout(this.timeout);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({ order: nextProps.order });

    this.timeout = setTimeout(() => {
      this.props.fetchOrder(this.props.match.params.orderRef);
    }, config.ORDER_DETAILS_FETCH_INTERVAL);

    if (nextProps.order !== 429) {
      this.setState({ order: nextProps.order });

      if (
        this.props.order &&
        this.props.order.status_name.length > 0 &&
        this.props.order.status_name[0][0] === 11 &&
        nextProps.order.status_name[0][0] === 12
      ) {
        window.ga('send', 'event', 'Order', 'order paid', nextProps.unique_reference);
      }
    }
  }

  render() {
    if (this.state.order == null) {
      return <Loading />;
    } else if (this.state.order === 404) {
      return <NotFound />;
    } else if (typeof this.state.order === 'object') {
      return (
        <div id="order" className={isFiatOrder(this.state.order) ? 'order-fiat' : 'order-crypto'}>
          <div className="container">
            <OrderTop order={this.state.order} />

            <div className="row">
              <CoinProcessed type="Deposit" order={this.state.order} />
              <CoinProcessed type="Receive" order={this.state.order} />

              <OrderInfo order={this.state.order} />
              <Notifications order={this.state.order} />

              {!isFiatOrder(this.state.order) && <RefundAddress order={this.state.order} />}

              <ReferralBox order={this.state.order} />
            </div>
          </div>
        </div>
      );
    }
  }
}

const mapStateToProps = ({ order }) => ({ order });

export default connect(mapStateToProps, { fetchOrder })(Order);
