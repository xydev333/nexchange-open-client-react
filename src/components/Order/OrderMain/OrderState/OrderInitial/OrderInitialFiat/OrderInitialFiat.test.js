import React from 'react';
import { shallow } from 'enzyme';
import OrderInitialFiat from './OrderInitialFiat';
import order from 'Mocks/order';

describe('OrderInitialFiat', () => {
  it('renders correctly', () => {
    expect(shallow(<OrderInitialFiat order={order} />)).toMatchSnapshot();
  });
});
