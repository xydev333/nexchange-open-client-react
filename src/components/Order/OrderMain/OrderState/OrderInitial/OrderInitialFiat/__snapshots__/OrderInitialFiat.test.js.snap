// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderInitialFiat renders correctly 1`] = `ShallowWrapper {}`;
